src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -136,7 +136,9 @@ private bool IsDescendantOf(SchemaObjectType schemaObjectType)
                 return false;
             var descendantTypes = schemaObjectType.Discriminator.Implementations.Select(implementation => implementation.Type).ToHashSet();
 
-            return descendantTypes.Contains(Type);
+            return descendantTypes.Contains(Type) ||
+                // We need this redundant check as the internal backing schema will not be a part of the discriminator implementations of its base type.
+                ObjectSchema.Parents?.All.Count == 1  && ObjectSchema.Parents.All.First().Equals(schemaObjectType.ObjectSchema);
```

Comment:
is it possible that we could combine this extra check into the descendantTypes?
Or to make the procedure clear, if this check is working as an assertion, we should check it at the first of this method like
```csharp
if (ObjectSchema.Parents?.All.Count != 1 || !ObjectSchema.Parents.All.First().Equals(schemaObjectType.ObjectSchema))
     return false;
```

Reply:
Fixed, I changed the code and will add the type into descendantTypes if the type is used to support forward compatibility in polymorphism.